indexer/src/main/scala/scalding/BaseFeaturesImporterJob.scala

```diff
@@ -0,0 +1,191 @@
+      val wktReader = new WKTReader()
+      val wkbWriter = new WKBWriter()
+      wkbWriter.write(wktReader.read(polygon))
+    })
+  }
+
+  private def getEmbeddedBoundingBox(feature: GeonamesFeature): Option[BoundingBox] = {
+    feature.extraColumns.get("bbox").flatMap(bboxStr => {
+      // west, south, east, north
+      val parts = bboxStr.split(",").map(_.trim)
+      parts.toList match {
+        case w :: s :: e :: n :: Nil => {
+          Some(BoundingBox(Point(n.toDouble, e.toDouble), Point(s.toDouble, w.toDouble)))
+        }
+        case _ => {
+          // logger.error("malformed bbox: " + bboxStr)
+          None
+        }
+      }
+    })
+  }
+
+  private def getEmbeddedBoost(feature: GeonamesFeature): Option[Int] = {
+    feature.extraColumns.get("boost").map(_.toInt)
+  }
+
+  val countryInfoCachedFile = getCachedFileByRelativePath("downloaded/countryInfo.txt")
+  val adminCodesCachedFile = getCachedFileByRelativePath("downloaded/adminCodes.txt")
+  @transient lazy val adminCodeMap = InMemoryLookupTableHelper.buildAdminCodeMap(
+    countryInfoCachedFile,
+    adminCodesCachedFile)
+
+  private def getEmbeddedParents(feature: GeonamesFeature): List[Long] = {
+    def lookupAdminCode(p: String): Option[String] = {
+      adminCodeMap.get(p)
+    }
+
+    val explicitParents = for {
+      parentIdStrings <- feature.extraColumns.get("parents").toList
+      fidString <- parentIdStrings.split(",").toList
+      fid <- StoredFeatureId.fromHumanReadableString(fidString, Some(GeonamesNamespace))
+    } yield {
+      fid.longId
+    }
+
+    val adminCodeParents = for {
+      parentAdminCode <- feature.parents.toList
+      fidString <- lookupAdminCode(parentAdminCode)
+      fid <- StoredFeatureId.fromHumanReadableString(fidString, Some(GeonamesNamespace))
+    } yield {
+      fid.longId
+    }
+
+    (explicitParents ++ adminCodeParents).distinct
+  }
+
+  private def getEmbeddedCanGeocode(feature: GeonamesFeature): Boolean = {
+    feature.extraColumns.get("canGeocode").map(_.toInt).getOrElse(1) > 0
+  }
+
+  private def getEmbeddedAttributes(feature: GeonamesFeature): Option[GeocodeFeatureAttributes] = {
+    var attributesSet = false
+    lazy val attributesBuilder = {
+      attributesSet = true
+      GeocodeFeatureAttributes.newBuilder
+    }
+
+    if (feature.featureClass.isAdmin1Capital) {
+      attributesBuilder.adm1cap(true)
+    }
+
+    feature.population.foreach(pop =>
+      attributesBuilder.population(pop)
+    )
+
+    feature.extraColumns.get("sociallyRelevant").map(v =>
+      attributesBuilder.sociallyRelevant(v.toBoolean)
+    )
+
+    feature.extraColumns.get("neighborhoodType").map(v =>
+      attributesBuilder.neighborhoodType(NeighborhoodType.findByNameOrNull(v))
+    )
+
+    if (attributesSet) {
+      Some(attributesBuilder.result)
+    } else {
+      None
+    }
+  }
+
+  (for {
+    line <- lines
+    feature <- lineProcessor(0, line)
+    // TODO: support config.shouldParseBuildings
```

Comment:
That can be either passed in `args` or via a `org.apache.hadoop.conf.Configuration` somewhere?